src/extension/background-script/state.js

```diff
@@ -0,0 +1,62 @@
+import browser from "webextension-polyfill";
+import createState from "zustand";
+
+// these keys get synced from the state to the browser storage
+// the values are the default values
+const browserStorage = {
+  settings: {},
+  accounts: {},
+  allowances: [],
+  currentAccountId: null,
+};
+
+const state = createState((set, get) => ({
+  connector: null,
+  account: null,
+  settings: {},
+  accounts: {},
+  allowances: [],
+  currentAccountId: null,
+  password: null,
+  lock: () => set({ password: null, connector: null, account: null }),
+  init: () => {
+    return browser.storage.sync
+      .get(Object.keys(browserStorage))
+      .then((result) => {
+        const data = {};
+        Object.keys(browserStorage).forEach((key) => {
+          data[key] = result[key] || browserStorage[key];
+        });
+        set(data);
+      });
+  },
+  flush: () => {
```

Comment:
What is this and what does it do?

Reply:
currently it is not used. was experimenting with how to actively write the object to the browser storage